tests/test_issue_maker.py

```diff
@@ -0,0 +1,161 @@
+
+    def test_that_title_mismatches(self):
+        from swaglyrics_backend.issue_maker import is_title_mismatched
+        self.assertTrue(
+            is_title_mismatched({"Bohemian", "Rhapsody", "by", "Queen"}, "Miracle by Caravan Palace", 2))
+
+    def test_that_title_not_mismatches(self):
+        from swaglyrics_backend.issue_maker import is_title_mismatched
+        self.assertFalse(
+            is_title_mismatched({"Bohemian", "Rhapsody", "by", "Queen"}, "Bohemian Rhapsody by Queen", 2))
+
+    def test_that_title_not_mismatches_with_one_error(self):
+        from swaglyrics_backend.issue_maker import is_title_mismatched
+        self.assertFalse(is_title_mismatched({"Bohemian", "Rhapsody", "2011", "by", "Queen"}, "Bohemian Rhapsody "
+                                                                                              "by Queen", 2))
+
+    @patch('swaglyrics_backend.issue_maker.db')
+    def test_that_add_stripper_adds_stripper(self, app_mock):
+        """
+        This test doesn't test database behaviour! Only dealing with unsupported and parsing request
+        """
+        from swaglyrics_backend.issue_maker import app, add_stripper
+        with app.test_client() as c:
+            c.post('/add_stripper', data={'auth': '',
+                                          'song': 'Miracle',
+                                          'artist': 'Caravan Palace',
+                                          'stripper': 'Caravan-palace-miracle'})
+            generate_fake_unsupported()
+            result = add_stripper()
+            self.assertEqual(f"Added stripper for Miracle by Caravan Palace to server database successfully, "
+                             f"deleted 1 instances from "
+                             "unsupported.txt", result)
+
+    def test_that_master_unsupported_reads_data(self):
+        from swaglyrics_backend.issue_maker import app
+        with app.test_client() as c:
+            generate_fake_unsupported()
+            req = c.get('/master_unsupported')
+            self.assertIsNotNone(req.response)
+
+    def test_that_delete_line_deletes_line_from_master_unsupported(self):
+        from swaglyrics_backend.issue_maker import app, delete_line
+        with app.test_client() as c:
+            c.post('/delete_unsupported', data={'auth': '',
+                                                'song': 'Supersonics',
+                                                'artist': 'Caravan Palace'})
+            generate_fake_unsupported()
+            response = delete_line()
+            self.assertEqual(response, "Removed 1 instances of Supersonics by Caravan Palace from "
+                                       "unsupported.txt successfully.")
+
+
+def get_correct_spotify_search_json(filename):
+    import flask
+    with open(filename, 'r') as r:
+        raw_json = r.read()
+        return flask.json.loads(raw_json)
+
+
+def generate_fake_unsupported():
```

Comment:
this uses the same unsupported.txt or creates a new one in tests folder?

Reply:
New one in tests folder

Reply:
That's good